tests/test_threecomponents_ptflash.cpp

```diff
@@ -0,0 +1,139 @@
+
+    fluid_state.setMoleFraction(FluidSystem::oilPhaseIdx, FluidSystem::Comp0Idx, comp[0]);
+    fluid_state.setMoleFraction(FluidSystem::oilPhaseIdx, FluidSystem::Comp1Idx, comp[1]);
+    fluid_state.setMoleFraction(FluidSystem::oilPhaseIdx, FluidSystem::Comp2Idx, comp[2]);
+
+    fluid_state.setMoleFraction(FluidSystem::gasPhaseIdx, FluidSystem::Comp0Idx, comp[0]);
+    fluid_state.setMoleFraction(FluidSystem::gasPhaseIdx, FluidSystem::Comp1Idx, comp[1]);
+    fluid_state.setMoleFraction(FluidSystem::gasPhaseIdx, FluidSystem::Comp2Idx, comp[2]);
+
+    // It is used here only for calculate the z
+    fluid_state.setSaturation(FluidSystem::oilPhaseIdx, sat[0]);
+    fluid_state.setSaturation(FluidSystem::gasPhaseIdx, sat[1]);
+
+    fluid_state.setTemperature(temp);
+
+    // ParameterCache paramCache;
+    {
+        typename FluidSystem::template ParameterCache<Evaluation> paramCache;
+        paramCache.updatePhase(fluid_state, FluidSystem::oilPhaseIdx);
+        paramCache.updatePhase(fluid_state, FluidSystem::gasPhaseIdx);
+        fluid_state.setDensity(FluidSystem::oilPhaseIdx, FluidSystem::density(fluid_state, paramCache, FluidSystem::oilPhaseIdx));
+        fluid_state.setDensity(FluidSystem::gasPhaseIdx, FluidSystem::density(fluid_state, paramCache, FluidSystem::gasPhaseIdx));
+    }
+
+    ComponentVector z(0.); // TODO; z needs to be normalized.
+    {
+        Scalar sumMoles = 0.0;
+        for (unsigned phaseIdx = 0; phaseIdx < FluidSystem::numPhases; ++phaseIdx) {
+            for (unsigned compIdx = 0; compIdx < numComponents; ++compIdx) {
+                Scalar tmp = Opm::getValue(fluid_state.molarity(phaseIdx, compIdx) * fluid_state.saturation(phaseIdx));
+                z[compIdx] += Opm::max(tmp, 1e-8);
+                sumMoles += tmp;
+            }
+        }
+        z /= sumMoles;
+        // p And z is the primary variables
+        Evaluation z_last = 1.;
+        for (unsigned compIdx = 0; compIdx < numComponents - 1; ++compIdx) {
+            z[compIdx] = Evaluation::createVariable(Opm::getValue(z[compIdx]), compIdx + 1);
+            z_last -= z[compIdx];
+        }
+        z[numComponents - 1] = z_last;
+    }
+
+    const double flash_tolerance = 1.e-12; // just to test the setup in co2-compositional
+    const int flash_verbosity = 1;
+    const std::string flash_twophase_method = "newton";
+
+    // TODO: should we set these?
+    // Set initial K and L
+    for (unsigned compIdx = 0; compIdx < numComponents; ++compIdx) {
+        const Evaluation Ktmp = fluid_state.wilsonK_(compIdx);
+        fluid_state.setKvalue(compIdx, Ktmp);
+    }
+    const Evaluation Ltmp = 1.;
+    fluid_state.setLvalue(Ltmp);
+
+    const int spatialIdx = 0;
+    using Flash = Opm::PTFlash<double, FluidSystem>;
+    Flash::solve(fluid_state, z, spatialIdx, flash_verbosity, flash_twophase_method, flash_tolerance);
+
+}
+
+int main(int argc, char **argv)
+{
+    Dune::MPIHelper::instance(argc, argv);
+
+    testPTFlash();
```

Comment:
Is this test restricted to "compile + run" only?

Reply:
I added some checking of the values and derivatives after the flash calculation, so the results of the flash calculation is checked.